dev/Notifications/BadgeNotifications/BadgeNotificationManager.cpp

```diff
@@ -0,0 +1,157 @@
+    using namespace ::ABI::Windows::Foundation::Collections;
+}
+
+namespace BaseNotifications
+{
+    using namespace ::Microsoft::Windows::BaseNotifications;
+}
+
+namespace winrt::Microsoft::Windows::BadgeNotifications::implementation
+{
+    BadgeNotificationManager::BadgeNotificationManager() : BaseNotifications::BaseNotificationManager() {}
+
+    winrt::Microsoft::Windows::BadgeNotifications::BadgeNotificationManager BadgeNotificationManager::Current()
+    {
+        auto appProperties{ winrt::CoreApplication::Properties() };
+
+        static wil::srwlock lock;
+
+        auto criticalSection{ lock.lock_exclusive() };
+        auto storedBadgeNotificationManager{ appProperties.TryLookup(STORED_BADGENOTIFICATION_MANAGER_KEY) };
+        if (storedBadgeNotificationManager)
+        {
+            return storedBadgeNotificationManager.as<winrt::Microsoft::Windows::BadgeNotifications::BadgeNotificationManager>();
+        }
+        else
+        {
+            // Need to clear the RoActivateInstance caching for the PushNotificationLongRunningProcess proxyStub to be found.
+            ::Microsoft::RoApi::ClearRoActivateInstanceCache();
+
+            // Store the BadgeNotificationManager in the COM static store
+            auto badgeNotificationManager{ winrt::make<BadgeNotificationManager>() };
+            appProperties.Insert(STORED_BADGENOTIFICATION_MANAGER_KEY, badgeNotificationManager);
+            return badgeNotificationManager;
+        }
+    }
+
+    void BadgeNotificationManager::SetBadgeAsCount(uint32_t notificationCount)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        SetBadge(std::to_wstring(notificationCount), nullptr);
+        return;
+    }
+
+    void BadgeNotificationManager::SetBadgeAsCount(uint32_t notificationCount, winrt::DateTime expiration)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        SetBadge(std::to_wstring(notificationCount), &expiration);
+        return;
+    }
+
+    void BadgeNotificationManager::SetBadgeAsGlyph(winrt::BadgeNotificationGlyph glyphValue)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        PCWSTR glyphValueString;
+        GetBadgeNotificationGlyphToString(glyphValue, &glyphValueString);
+        SetBadge(glyphValueString, nullptr);
+        return;
+    }
+
+    void BadgeNotificationManager::SetBadgeAsGlyph(winrt::BadgeNotificationGlyph glyphValue, winrt::DateTime expiration)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        PCWSTR glyphValueString;
+        GetBadgeNotificationGlyphToString(glyphValue, &glyphValueString);
```

Comment:
handle winrt::BadgeNotificationGlyph::None case, call clear method ?

Reply:
When we give None, shell automatically does not display the notification. Same behavior as WindowsSdk

Reply:
As per spec .. when this function is called with None, the notifications should be cleared. 
consider calling clear() in this case.